templates/terraform/encoders/bigtable_app_profile.go.erb

```diff
@@ -0,0 +1,19 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2020 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+<%# Because instance is a URL param only, it does not get expanded and
+  # the URL is constructed from ResourceData. Set it in
+  # state and use a encoder instead of a field expander -%>
+// Instance is a URL parameter only, so replace self-link/path with resource name only.
+d.Set("instance", GetResourceNameFromSelfLink(d.Get("instance").(string)))
```

Comment:
Looks like encoders have to return the json map and an err? Probably just the object passed in and nil in this case

Reply:
herp